src/dmd/console.d

```diff
@@ -97,21 +119,54 @@ struct Console
             if (GetConsoleScreenBufferInfo(h, &sbi) == 0) // get initial state of console
                 return null;
 
+            // try enabling support for ANSI escape codes
+            DWORD mode;
+            bool vtmode;
+            bool resetVtmode;
+            if (GetConsoleMode(h, &mode))
+            {
+                if (mode & ENABLE_VIRTUAL_TERMINAL_PROCESSING)
+                {
+                    // the flag was enabled already, do not touch it afterwards
+                    vtmode = true;
+                }
+                else if (SetConsoleMode(h, mode | ENABLE_VIRTUAL_TERMINAL_PROCESSING))
```

Comment:
Why?

Furthermore, if the compiler crashes, we will have changed the setting of something with a longer lifetime that the compiler (the console window).

Reply:
I don't think it's required to turn off the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` flag, but I thought it would be common courtesy turn off something we enabled.